### ISSUE TITLE
About the SemanticConnectivityLoss

### ISSUE THREAD

**Issue by dreamer121121** [PaddlePaddle/PaddleSeg#1751]
Thanks for your issue. To help us solve the issue better, please provide following information：
 1. PaddleSeg version: (please specify the branch as well，e.g. PaddleSeg release/2.3)
 2. PaddlePaddle version: (e.g. PaddlePaddle 2.1.0)
 3. Operation system: (e.g. Linux/Windows/MacOS)
 4. Python version: (e.g. Python3.6/7/8)
 5. CUDA/cuDNN version: (e.g. CUDA10.2/cuDNN 7.6.5)
 6. Additional context: (Add any other context about the problem)

---
In the paper: <<PP-HumanSeg: Connectivity-Aware Portrait Segmentationwith a Large-Scale Teleconferencing Video Dataset>>
The  SemanticConnectivityLoss has a special case when there is no pair exists between P and G, 0-connectivityin SCL would lead to zeros gradients. so, in this case, SCL loss should be treated differently, but i cannt find these code
欢迎您反馈PaddleSeg使用问题，辛苦您提供以下信息，方便我们快速定位和解决问题：
 1. PaddleSeg版本：（请提供版本号和分支信息，如PaddleSeg release/2.3）
 2. PaddlePaddle版本：（如PaddlePaddle 2.1.0）
 3. 操作系统信息：（如Linux/Windows/MacOS）
 4. Python版本号：（如Python3.6/7/8）
 5. CUDA/cuDNN版本：（ 如CUDA10.2/cuDNN 7.6.5等）
 6. 其他内容: （增加其他与问题相关的内容）

**Comment by LutaoChu** [comment]
https://github.com/PaddlePaddle/PaddleSeg/blob/1f49e5cda6f65e8d483d8e42b739f93feb7d4639/paddleseg/models/losses/semantic_connectivity_loss.py#L106

The corresponding code is here. It should be noted that it is time-consuming to calculate whether there is no pair between P and G, so we use a fast implementation here, that is, to judge whether there is no connected component between P and G as a substitute.

**Comment by rose-jinyang** [comment]
Hi @LutaoChu and @dreamer121121
In the forward function of the SemanticConnectivityLoss implementation(https://github.com/PaddlePaddle/PaddleSeg/blob/release/2.4/paddleseg/models/losses/semantic_connectivity_loss.py), are the parameter logits the probabilities map or logit map without applying softmax?
BTW:

![image](https://user-images.githubusercontent.com/47862419/154410883-6041effe-1b29-4a8e-888a-406d10299c2d.png)

There is an issue with the above function implementation.
pred_conn has labels greater than pred_num_conn value.
That's because pred_num_conn is limited as smaller than 10.

![image](https://user-images.githubusercontent.com/47862419/154411250-5145d709-64e6-4da6-b509-3f7c0ca8ae41.png)

This results in an issue in the below line.
pred_conn = F.one_hot(pred_conn.long(), pred_num_conn)

How can we fix this issue?

**Comment by LutaoChu** [comment]
refer to https://github.com/PaddlePaddle/PaddleSeg/issues/1798
